ci-operator/step-registry/fips-check/node-scan/fips-check-node-scan-commands.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'FRC=$?; create_fips_check_node_scan_junit' EXIT TERM
+
+# Generate the Junit for node scan
+function create_fips_check_node_scan_junit() {
+    echo "Generating the Junit for fips check node scan"
+    filename="fips-check-node-scan"
+    testsuite="fips-check-node-scan"
+    subteam="Security_and_Compliance"
+    if (( FRC == 0 )); then
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="0" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="${subteam}:Node scan of fips check should succeedded or skipped"/>
+</testsuite>
+EOF
+    else
+        cat >"${ARTIFACT_DIR}/${filename}.xml" <<EOF
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuite name="${testsuite}" failures="1" errors="0" skipped="0" tests="1" time="1">
+  <testcase name="${subteam}:Node scan of fips check should succeedded or skipped">
+    <failure message="">Fips node scan check failed</failure>
+  </testcase>
+</testsuite>
+EOF
+    fi
+}
+
+function set_proxy () {
+    if test -s "${SHARED_DIR}/proxy-conf.sh" ; then
+        echo "setting the proxy"
+        echo "source ${SHARED_DIR}/proxy-conf.sh"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "no proxy setting."
+    fi
+}
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+set_proxy
+run_command "oc whoami"
+run_command "oc version -o yaml"
+
+# skip for ARM64
+mapfile -t node_archs< <(oc get nodes -o jsonpath --template '{range .items[*]}{.status.nodeInfo.architecture}{"\n"}{end}' | sort -R | uniq -d)
+echo "The architectures included in the current cluster nodes are ${node_archs[*]}"
+NON_SUPPORTED_ARCHES=(arm64)
+for arch in "${node_archs[@]}"; do
+    if [[ "${NON_SUPPORTED_ARCHES[*]}" =~ $arch ]]; then
+        echo "Skip this test since it doesn't support $arch."
+        exit 0
+    fi
+done
+
+# get a master node
+master_node_0=`oc get node -l node-role.kubernetes.io/master= --no-headers | grep -Ev "NotReady|SchedulingDisabled"| awk '{print $1}' | awk 'NR==1{print}'`
+if [[ -z $master_node_0 ]]; then
+    echo "Error master node0 name is null!\n."
```

Comment:
`echo` does not understand `\n` unless `-e` option is given. If you really want to print a period on a separate like, this can be rewritten as:
```bash
echo "Error master node0 name is null!"
echo "."
```

or

```bash
printf  "Error master node0 name is null!\n.\n"